collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -186,12 +187,22 @@ public static String getCurrentFormIdentifierHash() {
 
     /**
      * Gets a unique, privacy-preserving identifier for a form based on its id and version.
-     * @param formId id of a form
+     *
+     * @param formId      id of a form
      * @param formVersion version of a form
      * @return md5 hash of the form title, a space, the form ID
      */
     public static String getFormIdentifierHash(String formId, String formVersion) {
-        String formIdentifier = new FormsDao().getFormTitleForFormIdAndFormVersion(formId, formVersion) + " " + formId;
+        Form form = new DatabaseFormsRepository().getLatestByFormIdAndVersion(formId, formVersion);
+
+        String formTitle;
+        if (form != null) {
```

Comment:
Don't you like the ternary operator? I would use it in cases like this but maybe you don't like it?

Reply:
Nope that'd definitely be better here. Will change.